rosbag2_cpp/src/rosbag2_cpp/action_utils.cpp

```diff
@@ -0,0 +1,152 @@
+  {TopicsInAction::Status, "^action_msgs/msg/GoalStatusArray$"}
+};
+
+const size_t kMinActionTopicPostfixLen = ActionTopicPostfix.at(TopicsInAction::Status).length();
+
+bool is_topic_related_to_action(const std::string & topic_name, const std::string & topic_type)
+{
+  TopicsInAction topic = TopicsInAction::Unknown;
+  if (topic_name.length() <= kMinActionTopicPostfixLen) {
+    return false;
+  } else {
+    for (auto &[topic_type_enum, postfix] : ActionTopicPostfix) {
+      if (topic_name.length() > postfix.length() &&
+        topic_name.compare(
+          topic_name.length() - postfix.length(), postfix.length(), postfix) == 0)
+      {
+        topic = topic_type_enum;
+        break;
+      }
+    }
+  }
+
+  if (topic == TopicsInAction::Unknown) {
+    return false;
+  }
+
+  std::regex pattern(ActionTopicTypeRegex.at(topic));
+  return std::regex_search(topic_type, pattern);
+}
+
+std::string action_topic_name_to_action_name(const std::string & topic_name)
+{
+  std::string action_name;
+  if (topic_name.length() <= kMinActionTopicPostfixLen) {
+    return action_name;
+  } else {
+    for (auto &[topic_type_enum, postfix] : ActionTopicPostfix) {
+      if (topic_name.length() > postfix.length() &&
+        topic_name.compare(
+          topic_name.length() - postfix.length(), postfix.length(), postfix) == 0)
+      {
+        action_name = topic_name.substr(0, topic_name.length() - postfix.length());
+        break;
+      }
+    }
+  }
+
+  return action_name;
+}
+
+std::string action_topic_type_to_action_type(const std::string & topic_type)
+{
+  std::string service_type;
+
+  for (auto &[topic_type_enum, regex] : ActionTopicTypeRegex) {
+    std::regex pattern(regex);
+    if (std::regex_search(topic_type, pattern)) {
+      switch (topic_type_enum) {
+        case TopicsInAction::SendGoalEvent:
+          // Remove the postfix "_SendGoal_Event"
+          service_type =
+            topic_type.substr(0, topic_type.length() - std::strlen("_SendGoal_Event"));
+          break;
+        case TopicsInAction::GetResultEvent:
+          // Remove the postfix "_GetResult_Event"
+          service_type =
+            topic_type.substr(0, topic_type.length() - std::strlen("_GetResult_Event"));
+          break;
+        case TopicsInAction::Feedback:
+          // Remove the postfix "_FeedbackMessage"
+          service_type =
+            topic_type.substr(0, topic_type.length() - std::strlen("_FeedbackMessage"));
+          break;
+        case TopicsInAction::CancelGoalEvent:
+        case TopicsInAction::Status:
+        default:
+          break;
+      }
+      return service_type;
+    }
+  }
+
+  return service_type;
+}
+
+TopicsInAction get_action_topic_type_from_topic_name(const std::string & topic_name)
+{
+  for (auto &[topic_type_enum, postfix] : ActionTopicPostfix) {
+    if (topic_name.length() > postfix.length() &&
+      topic_name.compare(
+        topic_name.length() - postfix.length(), postfix.length(), postfix) == 0)
```

Comment:
Nitpick: This could be in one line for the sake of better readability and supportability.
```suggestion
      topic_name.compare(topic_name.length() - postfix.length(), postfix.length(), postfix) == 0)
```